TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -84,6 +91,53 @@ public void Dispose()
         {
             if (_tcpClient.Connected)
                 _tcpClient.Close();
+        }
+
+        static TcpClient connectViaHTTPProxy(string targetHost, int targetPort,

```

Comment:
Please use `ConnectViaHttpProxy` as the name of the method, to respect .NET API design guidelines.